pkg/operator/cache/cache.go

```diff
@@ -36,81 +38,86 @@ var (
 // to limit what items end up in the cache. The following is removed/clear from the resources:
 // - pods -> managed fields, status (we care for spec to find out containers, the rests are set to empty)
 // - deploy/sts -> template.spec, status, managedfields (we only care about template/metadata except for injector deployment)
-func GetFilteredCache(podSelector labels.Selector) cache.NewCacheFunc {
-	cacheOptions := cache.Options{
-		TransformByObject: getTransformerFunctions(),
-	}
-
-	if podSelector != nil {
-		// The only pods we are interested are in watchdog. we don't need to list/watch pods that we are almost sure have dapr sidecar already
-		cacheOptions.SelectorsByObject = cache.SelectorsByObject{
-			&corev1.Pod{}: {
-				Label: podSelector,
-			},
+func GetFilteredCache(namespace string, podSelector labels.Selector) cache.NewCacheFunc {
+	return func(config *rest.Config, opts cache.Options) (cache.Cache, error) {
+		// The only pods we are interested are in watchdog. we don't need to
+		// list/watch pods that we are almost sure have dapr sidecar already.
+		opts.ByObject = getTransformerFunctions(podSelector)
+		if len(namespace) > 0 {
+			opts.DefaultNamespaces = map[string]cache.Config{
+				namespace: {},
```

Comment:
Not use namespace, is it really useful?

Reply:
Yes, this sets the namespaces which should be watched, [see here](https://github.com/kubernetes-sigs/controller-runtime/blob/2154ffbc22e26ffd8c3b713927f0df2fa40841f2/pkg/cache/cache.go#L193).